api/kogito-events-core/src/main/java/org/kie/kogito/event/cloudevents/utils/BaseCloudEventValidator.java

```diff
@@ -0,0 +1,133 @@
+ * limitations under the License.
+ */
+package org.kie.kogito.event.cloudevents.utils;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import io.cloudevents.SpecVersion;
+
+import static io.cloudevents.core.v1.CloudEventV1.DATACONTENTTYPE;
+import static io.cloudevents.core.v1.CloudEventV1.TIME;
+
+abstract class BaseCloudEventValidator {
+
+    private final String specVersionAttributeName;
+
+    private final SpecVersion supportedSpecVersion;
+
+    BaseCloudEventValidator(String specVersionAttributeName, SpecVersion supportedSpecVersion) {
+        this.specVersionAttributeName = specVersionAttributeName;
+        this.supportedSpecVersion = supportedSpecVersion;
+    }
+
+    final void validateCloudEvent(Map<String, Object> cloudEvent) throws InvalidCloudEventException {
+        Object specVersion = cloudEvent.get(specVersionAttributeName);
+
+        if (specVersion == null) {
+            throw new InvalidCloudEventException(List.of("Missing mandatory attribute: " + specVersionAttributeName));
+        }
+
+        if (specVersion.toString().equals(supportedSpecVersion.toString())) {
+            List<String> errors = new ArrayList<>();
+
+            CloudEventUtils.getMissingAttributes(cloudEvent)
+                    .forEach(missingAttribute -> errors.add("Missing mandatory attribute: " + missingAttribute));
+
+            errors.addAll(validateNonEmptyAttributes(cloudEvent));
+
+            errors.addAll(validateRfc2046Attributes(cloudEvent));
+
+            errors.addAll(validateRfc3339Attributes(cloudEvent));
+
+            if (!errors.isEmpty()) {
+                throw new InvalidCloudEventException(errors);
+            }
+        } else {
+            throw new UnsupportedOperationException(
+                    "Invalid CloudEvents specification version: " + specVersion + ". "
+                            + CloudEventValidatorV03.class.getName() + " supports only CloudEvents " + supportedSpecVersion);
+        }
+    }
+
+    /**
+     * Validates attributes that must adhere to the format specified in <a href="https://datatracker.ietf.org/doc/html/rfc3339">RFC 3339</a>.
+     *
+     * @param cloudEvent the CloudEvent
+     * @return a {@link List} of errors
+     */
+    private static List<String> validateRfc3339Attributes(Map<String, Object> cloudEvent) {
+        return Stream.of(TIME)
+                .filter(attribute -> {
+                    Object value = cloudEvent.get(attribute);
+                    return value != null && !isRfc3339Value(value);
+                }).map(attribute -> attribute + " MUST adhere to the format specified in RFC 3339 (https://datatracker.ietf.org/doc/html/rfc3339).")
+                .collect(Collectors.toList());
```

Comment:
There is only one attribute that isRfc3339. If there were more, then we should put them in a static set and iterate  
```suggestion
    String time = (String)cloudEvent.get(TIME);
    return isRfc3339Value(value) ? Collections.emptyList() : List.of(TIME + "" MUST adhere to the format specified in RFC 3339 (https://datatracker.ietf.org/doc/html/rfc3339)"); 
```